### PR TITLE
[shopsys] Redis upgrade

### PR DESCRIPTION
**Pull request by s3tezsky**
| Q             | A
| ------------- | ---
|Description, reason for the PR| There is newer version of redis client available. We want to keep our platform up to date.
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| Yes (upgrading dependency to new major version) 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

**Comment by grossmannmartin** [comment]
Hi, I rebased this pull request to the master branch as now the development version of 9.0.x release is on the master branch.

File: docs/installation/installation-using-docker-on-production-server.md [hunk 1]
```diff
@@ -158,7 +158,7 @@ firewall-cmd --permanent --zone=public --add-service=postgresql
 firewall-cmd --reload
 ```
 
-### Redis 4.0
+### Redis 5.0
```
Comment (TomasLudvik) [inline docs/installation/installation-using-docker-on-production-server.md:161]:
```suggestion
### Redis 5
```

**Comment by pesektomas** [comment]
> upgrade notes say about Redis server 5, but there is version 4 in docker-compose files.

It is simple. I am idiot, because I updated my local docker-compose file only. fb68210b579921dea9b3708dc6fafe3a5c932bd9

File: packages/framework/src/DependencyInjection/Compiler/LazyRedisCompilerPass.php [hunk 2]
```diff
@@ -24,5 +25,9 @@ public function process(ContainerBuilder $container): void
                 $definition->setLazy(true);
             }
         }
+
+        if ($container->getParameter('kernel.environment') === EnvironmentType::TEST) {
+            $container->getDefinition('snc_redis.test')->setPublic(true);
+        }
```
Comment (grossmannmartin) [inline packages/framework/src/DependencyInjection/Compiler/LazyRedisCompilerPass.php:31]: Is any particular reason to have the definition here in a compiler pass?
I would prefer to put it in `services_test.yml` if necessary.

Comment (grossmannmartin) [reply packages/framework/src/DependencyInjection/Compiler/LazyRedisCompilerPass.php:31]: Client service is created dynamically and is not possible to specify it in services_test file

**Pull request merged**